sechub-wrapper-prepare/src/main/java/com/mercedesbenz/sechub/wrapper/prepare/modules/PrepareWrapperModuleGit.java

```diff
@@ -0,0 +1,181 @@
+                    LOG.warn("User defined type as 'git', but the defined location was not a valid git location: {}", location);
+                    return false;
+                }
+                return true;
+            }
+
+            if (gitInputValidator.validateLocation(location)) {
+                LOG.debug("Location is a git URL");
+                return true;
+            }
+
+        }
+        return false;
+    }
+
+    public void prepare(PrepareWrapperContext context) throws IOException {
+
+        LOG.debug("Start remote data preparation for GIT repository");
+
+        List<SecHubRemoteDataConfiguration> remoteDataConfigurationList = context.getRemoteDataConfigurationList();
+
+        for (SecHubRemoteDataConfiguration secHubRemoteDataConfiguration : remoteDataConfigurationList) {
+            prepareRemoteConfiguration(context, secHubRemoteDataConfiguration);
+        }
+
+        if (!isDownloadSuccessful(context)) {
+            throw new IOException("Download of git repository was not successful.");
+        }
+        cleanup(context);
+    }
+
+    boolean isMatchingGitType(String type) {
+        if (type == null || type.isBlank()) {
+            return false;
+        }
+        return TYPE.equalsIgnoreCase(type);
+    }
+
+    boolean isDownloadSuccessful(PrepareWrapperContext context) {
+        // check if download folder contains git
+        Path path = Paths.get(context.getEnvironment().getPdsPrepareUploadFolderDirectory());
+        if (Files.isDirectory(path)) {
+            String gitFile = ".git";
+            Path gitPath = Paths.get(path + "/" + gitFile);
+            return Files.exists(gitPath);
+        }
+        return false;
+    }
+
+    private void prepareRemoteConfiguration(PrepareWrapperContext context, SecHubRemoteDataConfiguration secHubRemoteDataConfiguration) throws IOException {
+        String location = secHubRemoteDataConfiguration.getLocation();
+        Optional<SecHubRemoteCredentialConfiguration> credentials = secHubRemoteDataConfiguration.getCredentials();
+
+        if (!credentials.isPresent()) {
+            clonePublicRepository(context, location);
+            return;
+        }
+
+        Optional<SecHubRemoteCredentialUserData> optUser = credentials.get().getUser();
+        if (optUser.isPresent()) {
+            SecHubRemoteCredentialUserData user = optUser.get();
+            clonePrivateRepository(context, user, location);
+            return;
+        }
+
+        throw new IllegalStateException("Defined credentials have no credential user data for location: " + location);
+    }
+
+    private void clonePrivateRepository(PrepareWrapperContext context, SecHubRemoteCredentialUserData user, String location) throws IOException {
+        assertUserCredentials(user.getName(), user.getPassword());
+
+        HashMap<String, SealedObject> credentialMap = new HashMap<>();
+        addSealedUserCredentials(user.getName(), user.getPassword(), credentialMap);
```

Comment:
I am sorry for the confusion. I thought about passing the `SecHubRemoteCredentialUserData` called `user` to the methods below.
```suggestion
        assertUserCredentials(user);

        HashMap<String, SealedObject> credentialMap = new HashMap<>();
        addSealedUserCredentials(user, credentialMap);
```